### PR TITLE
CRM457-2198: Pricing

### PR DESCRIPTION
**Pull request by patrick-laa**
## Description of change
* Rather than do our own calculations that duplicate what the provider does, let them be done by the gem
* Rather than trust provider-submitted blob to find out the relevant rates, use the gem

[Link to relevant ticket](https://dsdmoj.atlassian.net/browse/CRM457-2198)

### CODE REVIEW

File: spec/mailers/nsm/feedback_messages/further_information_request_feeback_spec.rb [hunk 1]
```diff
@@ -12,7 +12,7 @@
   let(:ufn) { '123456/001' }
   let(:main_defendant_name) { 'Tracy Linklater' }
   let(:defendant_reference) { 'MAAT ID number: AB12123' }
-  let(:claim_total) { '£325.97' }
```
Comment (ivanELEC) [inline spec/mailers/nsm/feedback_messages/further_information_request_feeback_spec.rb:15]: Why did this value change?

**Review by ivanELEC — COMMENTED** [review]
In general the implementation is clear enough but confused about the tests with the values changing and also -1 being added as a placeholder (I think we discussed this part in a meeting but I forgot the reasoning, is it just not needed in the test but needed for the gem's input?)

File: spec/mailers/nsm/feedback_messages/further_information_request_feeback_spec.rb (hunk 1 above)
Comment (patrick-laa) [reply spec/mailers/nsm/feedback_messages/further_information_request_feeback_spec.rb:15]: In most of our tests our "pricing" values in the blob were unrealistic numbers, and we were using them to calculate the totals. Now we're not using values in the blob but rather actual pricing data from the gem, so the totals change.

**Comment by patrick-laa** [comment]
>  is it just not needed in the test but needed for the gem's input?

@ivanELEC  So our tests need to use data that matches the schema, but now there are a bunch of "deprecated" fields in the schema - i.e. places where we have been putting things like pricing, even though now we don't want to be reading those values from the blob (and when we introduce a v2 schema we'll remove them entirely). By having technically-valid-but-absolutely-loopy values like `-1` in those fields in our test data, it allowed me to verify that we weren't naughtily continuing to use those values in the code - if we were, the tests would have blown up with negative figures.

**Review by elken — APPROVED** [review]

**Pull request merged**